### PR TITLE
Add Appearing/Disappearing to root page and NavigationPage

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

- Add a SendAppearing event inside window. Currently this isn't completely correct but we'll just connect it up correctly once we have life cycle events inside window working properly
- I setup appearing/disappearing inside NavigationPage so that we have xplat consistency

### CODE REVIEW

File: src/Controls/src/Core/NavigationPage.cs [hunk 1]
```diff
@@ -401,11 +409,26 @@ async Task PopToRootAsyncInner(bool animated)
...
+				page?.SendAppearing();
+		}
+
 		async Task PushAsyncInner(Page page, bool animated)
 		{
 			if (InternalChildren.Contains(page))
 				return;
 
+			FireDisappearing(CurrentPage);
+			FireAppearing(page);
```
[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/src/Core/NavigationPage.cs:430]: No, appearing means this page has the intention of appearing so you need to start wiring up events 

Once we add NavigatedTo and NavigatedFrom those will fire after the stack has settled

File: src/Controls/src/Core/NavigationPage.cs [hunk 2]
```diff
@@ -293,23 +294,27 @@ protected override bool OnBackButtonPressed()
...
-			if (StackDepth == 1)
+			if (NavigationPageController.StackDepth == 1)
 			{
 				return null;
 			}
 
+			FireDisappearing(page);
+
+			if(InternalChildren.Last() == page)
+				FireAppearing((Page)InternalChildren[NavigationPageController.StackDepth - 2]);
```
[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/src/Core/NavigationPage.cs:316]: No, appearing means this page has the intention of appearing so you need to start wiring up events 

Once we add NavigatedTo and NavigatedFrom those will fire after the stack has settled

File: src/Controls/samples/Controls.Sample/Pages/SemanticsPage.xaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
                    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
                    xmlns:controls="clr-namespace:Maui.Controls.Sample.Controls"
                    x:Class="Maui.Controls.Sample.Pages.SemanticsPage"
-                   BackgroundColor="White">
```
[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/samples/Controls.Sample/Pages/SemanticsPage.xaml:5]: Changed back

File: src/Controls/src/Core/HandlerImpl/Window.Impl.cs [hunk 4]
```diff
@@ -35,6 +41,12 @@ void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 				foreach (Element item in e.NewItems)
 				{
 					OnChildAdded(item);
+
+					// TODO once we have better life cycle events on pages 
+					if (item is Page)
+					{
+						SendWindowAppearing();
```
[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/src/Core/HandlerImpl/Window.Impl.cs:48]: Can you point me to where in the code we'd do this?

[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Controls/src/Core/HandlerImpl/Window.Impl.cs:48]: I think this is fine for now

Once someone tackles life cycle events on the window we can wire this call into that but for now with one window I think this takes the least opinionated/functional approach

**Review by Redth — APPROVED** [review]

**Pull request merged**